### PR TITLE
add pagination in files items view

### PR DESCRIPTION
**Pull request by nettashviro**

### CODE REVIEW

File: src/api/files.js [hunk 1]
```diff
@@ -27,11 +27,23 @@ export async function getFoldersByFolder(parent) {
   }
 }
 
+/**
+ * fetchSharedFiles fetch all the shared folders in the current parent
+ * @param parent the file id of the parent
+ */
+export async function fetchSharedFolders(parent) {
+  const res = await Axios.get(`${baseURL}/api/files?shares&appId=drive${parent ? `&parent=${parent}` : ""}`);
+  const permissions = res.data.files.filter((item) => item.type == fileTypes.folder);
```
Comment (yoavrozner) [inline src/api/files.js:36]:
```suggestion
  const permissions = res.data.files.filter((item) => isFolder(item.type));
```

**Review by yoavrozner — CHANGES_REQUESTED** [review]

File: src/api/files.js (hunk 1 above)
Comment (nettashviro) [reply src/api/files.js]: done :)

File: src/components/files/preview/FilesPreview.vue [hunk 2]
```diff
@@ -69,11 +90,37 @@ export default {
       this.$emit("fileclick", file);
     },
   },
+  updated() {
+    if (!this.isListenerAttached) {
+      const filesElm = document.querySelector("#files-items-pagination");
```
Comment (yoavrozner) [inline src/components/files/preview/FilesPreview.vue:95]: I don't like it. Vue should have this kind of stuff

File: src/components/popups/info-popup/BaseInfoPopup.vue [hunk 3]
```diff
@@ -63,14 +63,16 @@ export default {
   },
   methods: {
     async open() {
-      this.users = await getPermissions(this.file.id);
+      let externalPermissionsRes;
+      [this.users, externalPermissionsRes] = await Promise.all([
+        getPermissions(this.file.id),
+        getExternalPermissions(this.file.id).catch(() => (externalPermissionsRes = [])),
```
Comment (yoavrozner) [inline src/components/popups/info-popup/BaseInfoPopup.vue:69]: Is it working on failure?

File: src/components/popups/info-popup/BaseInfoPopup.vue [hunk 4]
```diff
@@ -63,14 +63,16 @@ export default {
...
-      let externalPermissionsRes = await getExternalPermissions(this.file.id);
-
-      externalPermissionsRes = externalPermissionsRes.slice().sort((a, b) => b.createdAt - a.createdAt);
-      externalPermissionsRes = externalPermissionsRes.filter(
-        (user, index, self) => index === self.findIndex((anotherUser) => anotherUser.id === user.id)
-      );
+      externalPermissionsRes = externalPermissionsRes
+        .slice()
+        .sort((a, b) => b.createdAt - a.createdAt)
+        .filter((user, index, self) => index === self.indexOf(user.id));
```
Comment (yoavrozner) [inline src/components/popups/info-popup/BaseInfoPopup.vue:75]: Are you sure it is working well?

File: src/views/SharedWithMe.vue [hunk 5]
```diff
@@ -11,20 +10,47 @@
...
+      }
+    },
   },
   created() {
-    this.$store.dispatch("fetchSharedFiles", 0);
+    this.$store.dispatch("fetchSharedFiles", { pageNum: 0, pageAmount: this.getPageSize(this.fileView) });
   },
   methods: {
+    getPageSize(fileView) {
+      return fileView == 0 || fileView == "undefined" || fileView == undefined ? pageSize : pageSize * 2;
```
Comment (yoavrozner) [inline src/views/SharedWithMe.vue:36]: `fileView == "undefined" || fileView == undefined` don't you have a better way?

Comment (yoavrozner) [inline src/views/SharedWithMe.vue:36]: `fileView === 0`

**Review by yoavrozner — CHANGES_REQUESTED** [review]
Well done!

File: src/views/SharedWithMe.vue (hunk 5 above)
Comment (nettashviro) [reply src/views/SharedWithMe.vue:36]: how? it's different types

File: src/views/SharedWithMe.vue [hunk 6]
```diff
@@ -11,20 +10,47 @@
...
 export default {
   name: "Shared",
   components: { PageTemplate },
   computed: {
-    ...mapGetters(["files", "serverFilesLength"]),
+    ...mapGetters(["files", "serverFilesLength", "fileView", "currentFolder"]),
+  },
+  watch: {
+    fileView: function(newValue, oldValue) {
+      if (oldValue != newValue) {
```
[the remark this answers is not in the document]
Comment (nettashviro) [reply src/views/SharedWithMe.vue:23]: done

File: src/components/popups/info-popup/BaseInfoPopup.vue (hunk 3 above)
Comment (nettashviro) [reply src/components/popups/info-popup/BaseInfoPopup.vue:69]: done

File: src/components/files/preview/FilesPreview.vue (hunk 2 above)
Comment (nettashviro) [reply src/components/files/preview/FilesPreview.vue:95]: odne

**Review by Shahar-Y — APPROVED** [review]
Needs to be approved by @yoavrozner

File: src/components/files/preview/FilesPreview.vue [hunk 7]
```diff
@@ -69,11 +90,37 @@ export default {
       this.$emit("fileclick", file);
     },
   },
+  updated() {
+    if (!this.isListenerAttached) {
```
[the remark this answers is not in the document]
Comment (yoavrozner) [reply src/components/files/preview/FilesPreview.vue:94]: ?

File: src/components/popups/info-popup/BaseInfoPopup.vue [hunk 8]
```diff
@@ -63,23 +63,27 @@ export default {
...
+        externalPermissionsRes = externalPermissionsRes
+          .slice()
+          .sort((a, b) => b.createdAt - a.createdAt)
+          .filter((user, index, self) => index === self.indexOf(user.id));
+
+        externalPermissionsRes.forEach((externalPermission) => {
+          externalPermission.isFailed
+            ? this.externalUsersFailed.push(externalPermission)
+            : this.externalUsers.push(externalPermission);
+        });
```
Comment (yoavrozner) [inline src/components/popups/info-popup/BaseInfoPopup.vue:85]: Wouldn't `map` or `reduce` fit better here?

**Review by yoavrozner — COMMENTED** [review]
Great job @nettashviro !
Please see my notes :)

File: src/views/SharedWithMe.vue (hunk 5 above)
Comment (yoavrozner) [reply src/views/SharedWithMe.vue]: Uncaught TypeError: Cannot read property 'undefined' of null

File: src/components/popups/info-popup/BaseInfoPopup.vue (hunk 8 above)
Comment (nettashviro) [reply src/components/popups/info-popup/BaseInfoPopup.vue]: done

File: src/components/popups/info-popup/BaseInfoPopup.vue [hunk 9]
```diff
@@ -63,23 +64,28 @@ export default {
...
-      );
+      if (externalPermissionsRes) {
+        externalPermissionsRes = externalPermissionsRes
+          .slice()
+          .sort((a, b) => b.createdAt - a.createdAt)
+          .filter((user, index, self) => index === self.findIndex((anotherUser) => anotherUser.id === user.id));
 
-      this.externalUsers = [];
-      this.externalUsersFailed = [];
+        const [externalUsersFailed, externalUsers] = _.partition(externalPermissionsRes, function(externalPermission) {
```
Comment (yoavrozner) [inline src/components/popups/info-popup/BaseInfoPopup.vue:82]: Big like. I like this implementation. Thanks

**Review by yoavrozner — APPROVED** [review]
Well done! @nettashviro

**Pull request merged**